assets/Product.wxs

```diff
@@ -246,6 +249,17 @@
               </RegistryKey>
               <RemoveRegistryKey Id="RemoveOldOpenKey" Root="HKCR" Key="Directory\ContextMenus\$(var.ProductName)$(var.SimpleProductVersion)$(sys.BUILDARCH)\shell\open" Action="removeOnInstall"/>
             </Component>
+            <Component Id="PowerShellFileContextMenu">
+              <Condition>ADD_FILE_CONTEXT_MENU_RUNPOWERSHELL</Condition>
+              <!-- When right-clicking ps1 file -->
+              <RegistryKey Root="HKCR" Key="Microsoft.PowerShellScript.1\Shell\$(var.ProductName)$(var.SimpleProductVersion)$(sys.BUILDARCH)">
+                <RegistryValue Type="string" Name="MUIVerb" Value="$(var.PowerShellFileContextSubMenuDialogText)"/>
+                <RegistryValue Type="string" Name="Icon" Value="[VersionFolder]pwsh.exe"/>
```

Comment:
To avoid auto-picking key path, mark this one `KeyPath="yes"`.

Reply:
I will mark this `Icon` value as KeyPath, but I suppose any specific value element might be marked?

Reply:
I would choose that one that the MUIVerb contains potentially more variability but for the same path to pwsh.exe. You don't want the component generating a different GUID just for a display name while representing the same shared file.

Reply:
Thank you for the explanation. In commit 16aa665 element with Icon is marked as KeyPath.